packages/@lwc/synthetic-shadow/src/polyfills/iframe-content-window/patch.ts

```diff
@@ -54,9 +57,18 @@ function wrapIframeWindow(win: WindowProxy): WindowProxy {
             return win.focus.apply(win, arguments);
         },
         postMessage() {
-            // Typescript does not like it when you treat the `arguments` object as an array
-            // @ts-ignore type-mismatch
-            return win.postMessage.apply(win, arguments);
+            try {
+                // Typescript does not like it when you treat the `arguments` object as an array
+                // @ts-ignore type-mismatch
+                return win.postMessage.apply(win, arguments);
```

Comment:
this is a global lookup, in locker patches that method... it should be the patched method... which is already doing the exact same work, why is this needed here then?

Reply:
The `win` in this case is the `iframe.contentWindow` which there is no way for LWS to patch as `postMessage` is a bolt-on property to all window objects.  Since LWS does not see iframe.contentWindow's outside of a realm there is no way for us to tackle this. LWC on the other hand has this `iframe.contentWindow` wrapper that folks indirectly call into when passing objects out of the LWS realm.

Reply:
how come that win is an iframe? is LWC running on that iframe? if yes, why is not locker also running on that iframe?